### ISSUE TITLE
Inconsistent behavior for "read last 0 bytes" between REST and gRPC

### ISSUE THREAD

**Issue by coryan** [googleapis/google-cloud-cpp#9393]
I am not sure which one is the right behavior.  In gRPC you can issue a "read last 0 bytes" request, in REST a similar request returns ~all the bytes~ an error.  Maybe we should map all REST reads of "0 bytes" to "read the first 0 bytes"?  After all which empty string we return does not matter much?

**Comment by coryan** [comment]
Checked again.  With XML I get an error:

```
curl -i -H "Range: bytes=-0" -HHost:coryan-test-bucket.storage.googleapis.com -H "Authorization: Bearer $(gcloud auth print-access-token)" "https://storage.googleapis.com/quick-fox.txt"
HTTP/2 416 
x-guploader-uploadid: ADPycdvBS0-JZkIO7p-a3i616ouL6QKM_CiHsCL4vKPzjHV-b9qSOKOnSzsFMYFkid2tX0tiQtF87jgkeF_YY1w55ZMzNg
expires: Tue, 06 Sep 2022 22:37:15 GMT
date: Tue, 06 Sep 2022 22:37:15 GMT
cache-control: private, max-age=0
last-modified: Sun, 22 Aug 2021 19:38:20 GMT
etag: "9e107d9d372bb6826bd81d3542a419d6"
x-goog-generation: 1629661100240005
x-goog-metageneration: 1
x-goog-stored-content-encoding: identity
x-goog-stored-content-length: 43
x-goog-hash: crc32c=ImIEBA==
x-goog-hash: md5=nhB9nTcrtoJr2B01QqQZ1g==
x-goog-storage-class: MULTI_REGIONAL
accept-ranges: bytes
content-type: application/xml; charset=UTF-8
content-length: 164
server: UploadServer

<?xml version='1.0' encoding='UTF-8'?><Error><Code>InvalidRange</Code><Message>The requested range cannot be satisfied.</Message><Details>bytes=-0</Details></Error>
```

With REST I also get an error:

```
curl -i -H "Range: bytes=-0" -H "Authorization: Bearer $(gcloud auth print-access-token)" "https://storage.googleapis.com/storage/v1/b/coryan-test-bucket/o/quick-fox.txt?alt=media"
HTTP/2 416 
x-guploader-uploadid: ADPycdvs4t2wEgcAfI4rXXF1aJXegygvJPZWEVP0KHfvyfV_uhoM0p8Vmy0MdbAq6zjyEH6u7zr8TLdkijZVOBU-qNmG2vrxYI3h
content-range: bytes */43
content-type: text/html; charset=UTF-8
date: Tue, 06 Sep 2022 22:38:07 GMT
vary: Origin
vary: X-Origin
expires: Tue, 06 Sep 2022 22:38:07 GMT
cache-control: private, max-age=0
content-length: 29
server: UploadServer

Request range not satisfiable
```

**Comment by coryan** [comment]
The gRPC service cannot have the same behavior.  The `read_offset` is of type `int64`, it behaves identically for `0` and `-0`.  I need to fix in the gRPC client.


```
grpc_cli call --noremotedb --proto_path=/google/src/head/depot/google3/ --protofiles=google/storage/v2/storage.proto --channel_creds_type=gdc dns:///storage.googleapis.com:443  google.storage.v2.Storage.ReadObject 'bucket: "projects/_/buckets/cloud-cpp-testing-coryan" object: "fox.txt" read_offset: -0' 
connecting to dns:///storage.googleapis.com:443
Received initial metadata from server:
content-disposition : attachment
date : Tue, 06 Sep 2022 23:05:24 GMT
checksummed_data {
  content: "The quick brown fox jumps over the lazy dog\n"
  crc32c: 4264005234
}
object_checksums {
  crc32c: 4264005234
  md5_hash: "7\304\270~\337\374]\031\217\365\241\205\316\347\356\t"
}
metadata {
  name: "fox.txt"
  bucket: "projects/_/buckets/cloud-cpp-testing-coryan"
  generation: 1653408393546663
  metageneration: 1
  storage_class: "STANDARD"
  size: 44
  content_type: "application/octet-stream"
  create_time {
    seconds: 1653408393
    nanos: 677000000
  }
  checksums {
    crc32c: 4264005234
    md5_hash: "7\304\270~\337\374]\031\217\365\241\205\316\347\356\t"
  }
  update_time {
    seconds: 1653408393
    nanos: 677000000
  }
  update_storage_class_time {
    seconds: 1653408393
    nanos: 677000000
  }
  event_based_hold: false
}
Received trailing metadata from server:
content-disposition : attachment
Rpc succeeded with OK status
```

**Comment by coryan** [comment]
This was fixed in #4237.
